### PR TITLE
Modularization pt 2: Hidden modules

### PR DESCRIPTION
**Pull request by DStrand1**
Adds my module system, but without any user-facing differences.

This allows the modularization work to put all the code in its final home, but without a noticeable difference for players.

This also allows my module code to be reviewed by others.

Next PR:
- Commands

### CODE REVIEW

File: src/main/java/gregtech/modules/ModuleManager.java [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+            currentContainer = containers.get(getContainerID(module));
+            module.loadComplete(event);
+            module.getLogger().debug("Load-complete complete");
+        }
+    }
+
+    public void onServerStarting(FMLServerStartingEvent event) {
+        currentStage = ModuleStage.SERVER_STARTING;
+        for (IGregTechModule module : loadedModules) {
+            module.getLogger().debug("Server-starting start");
```
Comment (LAGIdiot) [inline src/main/java/gregtech/modules/ModuleManager.java:135]: CurrentContainer is not set here is it on purpose?

File: src/main/java/gregtech/api/modules/GregTechModule.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package gregtech.api.modules;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface GregTechModule {
```
Comment (LAGIdiot) [inline src/main/java/gregtech/api/modules/GregTechModule.java:10]: Where is this even used?

**Review by LAGIdiot — COMMENTED** [review]
There is quite lot of changes and it is quite hard to catch potential problems in them. Overall seems good but I found some places which needs at least another look or answers.

File: src/main/java/gregtech/api/modules/GregTechModule.java (hunk 2 above)
Comment (DStrand1) [reply src/main/java/gregtech/api/modules/GregTechModule.java:10]: Its used for module registration, done via ASM discovery (and then used periodically for Module metadata)

Comment (DStrand1) [reply src/main/java/gregtech/api/modules/GregTechModule.java:10]: See ModuleManager#configureModules

**Review by LAGIdiot — APPROVED** [review]
Implementation looks correct. Thanks for cleaning it up and making it more uniform.

**Pull request merged**